src/Core/OrganizationFeatures/AuthorizationHandlers/CollectionUserAuthorizationHandler.cs

```diff
@@ -0,0 +1,207 @@
+using Microsoft.AspNetCore.Authorization;
+
+namespace Bit.Core.OrganizationFeatures.AuthorizationHandlers;
+
+public class CollectionUserAuthorizationHandler : BulkAuthorizationHandler<CollectionUserOperationRequirement, CollectionUser>
+{
+    private readonly ICurrentContext _currentContext;
+    private readonly ICollectionRepository _collectionRepository;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+
+    public CollectionUserAuthorizationHandler(ICurrentContext currentContext, ICollectionRepository collectionRepository, IOrganizationUserRepository organizationUserRepository)
+    {
+        _currentContext = currentContext;
+        _collectionRepository = collectionRepository;
+        _organizationUserRepository = organizationUserRepository;
+    }
+
+    protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context,
+        CollectionUserOperationRequirement requirement,
+        ICollection<CollectionUser> resources)
+    {
+        switch (requirement)
+        {
+            case not null when requirement == CollectionUserOperation.Create:
+                await CanCreateAsync(context, requirement, resources);
+                break;
+            case not null when requirement == CollectionUserOperation.Delete:
+                await CanDeleteAsync(context, requirement, resources);
+                break;
+            case not null when requirement.Name == nameof(CollectionUserOperation.CreateForNewCollection):
+                await CanCreateForNewCollectionAsync(context, requirement, resources);
+                break;
+        }
+    }
+
+    /// <summary>
+    /// Ensure the acting user can create the requested <see cref="CollectionUser"/> resource(s).
+    /// </summary>
+    /// <remarks>
+    /// Checks that the following are all true:
+    /// - The target collection(s) exists 
+    /// - The acting user is an owner/admin AND the collection management setting is enabled
+    ///   OR
+    ///   The target collection(s) is manageable by the acting user. 
+    /// - The target user(s) exists and belongs to the same organization as the target collection
+    /// </remarks>
+    private async Task CanCreateAsync(AuthorizationHandlerContext context,
+        CollectionUserOperationRequirement requirement, ICollection<CollectionUser> resource)
+    {
+        if (!_currentContext.UserId.HasValue)
+        {
+            context.Fail();
+            return;
+        }
+
+        var distinctTargetCollectionIds = resource.Select(c => c.CollectionId).Distinct().ToList();
+
+        // List of collections the user is performing the operation on
+        var targetCollections = await _collectionRepository.GetManyByManyIdsAsync(distinctTargetCollectionIds);
+
+        // A target collection does not exist, fail the requirement
+        if (targetCollections.Count != distinctTargetCollectionIds.Count)
+        {
+            context.Fail();
+            return;
+        }
```

Comment:
It is somewhat redundant, but the `manageableCollections` query below excludes collections the user has no association with (i.e. via `CollectionUser` or `CollectionGroup`). So, the collection would come back as missing and fail the requirement. 

However, because there are organization settings that can allow users to modify access without a direct association with a collection; I needed to ensure they exist and belong to an organization that has those settings enabled or they're an Owner/Admin. 

And then, if no org settings are enabled and they're not an Owner/Admin, we need to continue and check for the `CanManage` setting for each targeted collection. 

We could possibly combine these query/repository calls into a single query that accepts multiple collection Ids _and_ a userId to also grab the permissions for that particular user. It would be an outer join though, that way if the permission is missing, the collection would still appear as existing for the organizational level checks. I opted to avoid that path for now to avoid introducing those new, more complicated query concepts at this time, but its definitely an area has potential for optimization.

Reply:
The path here might be a bit clearer after our latest discussion about the collection management settings. But from the discussion above it sounds like the flow needs to be something like:

1. get the `CurrentContentOrganization` (misspelling is intentional, should be `CurrentContextOrganization`) and check if the user is owner/admin for each relevant organization
    * you'll need to bring across this method from my PR: https://github.com/bitwarden/server/pull/2845/files#diff-a0d048d2dd3fd53b7dc8f8349cd348baf79c0ce075af29faff0a7abbc3683468R567-R570
    * this contains information about the user's permissions for that org, based on claims in the JWT - saves us a database call. CurrentContext will still construct that for us and make it available, but the actual business logic will be in handlers
3. if so, no further checks required for those orgs, we know they have permissions
4. do the `manageableCollections` db query + logic for the remaining collections, if any